cmd/swarmctl/network/common.go

```diff
@@ -14,31 +14,65 @@ import (
 func GetNetwork(ctx context.Context, c api.ControlClient, input string) (*api.Network, error) {
 	// GetService to match via full ID.
 	rg, err := c.GetNetwork(ctx, &api.GetNetworkRequest{NetworkID: input})
+
 	if err != nil {
-		// If any error (including NotFound), ListServices to match via ID prefix and full name.
-		rl, err := c.ListNetworks(ctx,
-			&api.ListNetworksRequest{
-				Filters: &api.ListNetworksRequest_Filters{
-					NamePrefixes: []string{input},
-				},
-			},
-		)
+		net, err := getNetworkByName(ctx, c, input)
 		if err != nil {
-			return nil, err
+			net, err = getNetworkByPrefixedID(ctx, c, input)
+			if err != nil {
+				return nil, err
+			}
 		}
+		return net, nil
+	}
 
-		if len(rl.Networks) == 0 {
-			return nil, fmt.Errorf("network %s not found", input)
-		}
+	return rg.Network, nil
+}
 
-		if l := len(rl.Networks); l > 1 {
-			return nil, fmt.Errorf("network %s is ambiguous (%d matches found)", input, l)
-		}
+func getNetworkByPrefixedID(ctx context.Context, c api.ControlClient, input string) (*api.Network, error) {
+	rl, err := c.ListNetworks(ctx,
+		&api.ListNetworksRequest{
+			Filters: &api.ListNetworksRequest_Filters{
+				IDPrefixes: []string{input},
+			},
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
 
-		return rl.Networks[0], nil
+	if len(rl.Networks) == 0 {
+		return nil, fmt.Errorf("network %s not found", input)
 	}
 
-	return rg.Network, nil
+	if l := len(rl.Networks); l > 1 {
+		return nil, fmt.Errorf("network %s is ambiguous (%d matches found)", input, l)
+	}
+
+	return rl.Networks[0], nil
+}
+
+func getNetworkByName(ctx context.Context, c api.ControlClient, input string) (*api.Network, error) {
+	rl, err := c.ListNetworks(ctx,
+		&api.ListNetworksRequest{
+			Filters: &api.ListNetworksRequest_Filters{
+				NamePrefixes: []string{input},
```

Comment:
I don't think this should use `NamePrefixes`. That introduces some additional ambiguity.